packages/devextreme/js/__internal/grids/new/grid_core/selection/controller.integration.test.ts

```diff
@@ -0,0 +1,174 @@
+import { throwError } from '@ts/grids/new/grid_core/options_validation/utils';
+import { rerender } from 'inferno';
+
+const SELECTORS = {
+  cardView: '.dx-cardview',
+  card: '.dx-cardview-card',
+  cardCheckbox: '.dx-checkbox-container',
+  selectAllButton: '[aria-label="Select all"]',
+};
+
+const setup = (options: GridCoreOptions = {}): CardView => {
+  const container = document.createElement('div');
+  const { body } = document;
+  body.append(container);
+
+  const cardView = new CardView(container, options);
+  rerender();
+  return cardView;
+};
+
+const getCardElements = (): NodeListOf<Element> => document.querySelectorAll(SELECTORS.card);
+
+const getCardCheckboxes = (): NodeListOf<Element> => document.querySelectorAll(SELECTORS.cardCheckbox);
+
+const getSelectAllButton = (): Element | null => document.querySelector(SELECTORS.selectAllButton);
+
+const checkError = (): void => expect(throwError).toHaveBeenCalledWith('E1042', 'CardView');
+
+jest.mock('@ts/grids/new/grid_core/options_validation/utils', () => ({
+  throwError: jest.fn().mockImplementation(() => ({})),
+}));
+
+describe('when keyExpr is missing', () => {
+  afterEach(() => {
+    const cardView = document.querySelector(SELECTORS.cardView);
+    // @ts-expect-error bad typed renderer
+    $(cardView ?? undefined as any)?.dxCardView('dispose');
+    document.body.innerHTML = '';
+  });
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+  describe('selection mode single', () => {
+    it('shouldn\'t throw E1042 error if keyExpr is missing and selection', () => {
+      setup({
+        dataSource: [{ value: 'test1' }, { value: 'test2' }],
+        selection: {
+          mode: 'single',
+        },
+      });
+
+      const cardElements = getCardElements();
+      expect(cardElements.length).toEqual(2);
+    });
+
+    it('should throw E1042 error on card click selection', () => {
+      setup({
+        dataSource: [{ value: 'test1' }, { value: 'test2' }],
+        selection: {
+          mode: 'single',
+        },
+      });
+
+      const cardElements = getCardElements();
+      cardElements[0].dispatchEvent(new MouseEvent('click'));
+
+      checkError();
+    });
+
+    it('should throw E1042 error on initial selectedCardKeys', () => {
+      setup({
+        dataSource: [{ value: 'test1' }, { value: 'test2' }],
+        selection: {
+          mode: 'single',
+        },
+        selectedCardKeys: [0],
+      });
+
+      checkError();
+    });
+
+    it('should throw E1042 error on runtime selectedCardKeys update', () => {
+      const cardView = setup({
+        dataSource: [{ value: 'test1' }, { value: 'test2' }],
+        selection: {
+          mode: 'single',
+        },
+      });
+
+      cardView.instance().option('selectedCardKeys', [1]);
+
+      checkError();
+    });
+  });
+
+  describe('selection mode multiple', () => {
+    it('shouldn\'t throw E1042 error if keyExpr is missing and selection', () => {
```

Comment:
Let's rename this test case
Now it looks like unfinished thought/sentence (and selection what? :))

E.g. 
```suggestion
    it('shouldn\'t throw E1042 on initial startup', () => {
```